src/main/java/ca/bc/gov/backendstartapi/entity/VegetationCode.java

```diff
@@ -0,0 +1,68 @@
+package ca.bc.gov.backendstartapi.entity;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.beans.Transient;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.ToString;
+import org.hibernate.annotations.NamedQuery;
+
+/**
+ * Code used to identify various vegetation species, along with metadata.
+ *
+ * <p>All date fields are stored as in the database as Oracle date types, which also store
+ * information about time down to seconds and without timezones (see <a
+ * href="https://stackoverflow.com/a/13568348">here</a>).
+ */
+@Entity
+@Table(name = "vegetation_code")
+@AllArgsConstructor
+@Getter
+@ToString
+@NamedQuery(
+    name = VegetationCode.FIND_VALID_BY_CODE_OR_DESCRIPTION,
+    query =
+        "SELECT vc FROM VegetationCode vc "
+            + "WHERE (vc.id ILIKE :search OR vc.description ILIKE :search) "
+            + "AND CURRENT_DATE >= vc.effectiveDate AND CURRENT_DATE < vc.expiryDate "
+            + "ORDER BY vc.id")
```

Comment:
Why don't have the query in the RepositoryImpl instead?

Reply:
Force of habit: I used to keep them in the entity class on my last job. Making it private inside the only class it would be used does seem a better idea.

Reply:
Turns out there is a reason for it, after all: [NamedQuery javadoc says](https://jakarta.ee/specifications/persistence/3.1/apidocs/jakarta.persistence/jakarta/persistence/namedquery) it "can be applied to an entity or mapped superclass".

Having a named query doesn't bring advantages right now, but the annotation allows us to [specify caching options and other things](https://jakarta.ee/specifications/persistence/3.1/apidocs/jakarta.persistence/jakarta/persistence/namedquery#hints()) in the future. Nothing we wouldn't be able to do if we just stored the string in the repository implementation: it's one option among many.